sensing/pointcloud_preprocessor/src/downsample_filter/pickup_based_voxel_grid_downsample_filter.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright 2023 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "pointcloud_preprocessor/downsample_filter/pickup_based_voxel_grid_downsample_filter.hpp"
+
+namespace
+{
+/**
+ * @brief Hash function for voxel keys.
+ * Utilizes prime numbers to calculate a unique hash for each voxel key.
+ */
+struct VoxelKeyHash
+{
+  std::size_t operator()(const std::array<int, 3> & k) const
+  {
+    // Primes based on the following paper: 'Investigating the Use of Primes in Hashing for
+    // Volumetric Data'.
+    return (k[0] * 73856093 ^ k[1] * 19349663 ^ k[2] * 83492791);
+    // return ((1 << 16) - 1) & (k[0] * 73856093 ^ k[1] * 19349663 ^ k[2] * 83492791);
+    // return ((1 << 20) - 1) & (k[0] * 73856093 ^ k[1] * 19349663 ^ k[2] * 83492791);
```

Comment:
Are these two lines necessary? 
If they are necessary, please add a comment to explain the purpose. If not, remove them. :pray: